viewing_party/main.py

```diff
@@ -0,0 +1,193 @@
+# WAVE ONE  
+
+def create_movie(title, genre, rating):
+    movie_dict = {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+    if title == None or genre == None or rating == None:
+        return None 
+    return movie_dict
+
+
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data 
+
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data
+    return user_data
+
+
+
+# WAVE TWO 
+
+    # PROBLEM ONE 
+    # I need to calculate the total sum of the ratings
+    # Then I need to divide the total sum by the length of the movies in the watched list 
+def get_watched_avg_rating(user_data):
+    sum_counter = 0
+    for movie in user_data["watched"]:
+        sum_counter += movie["rating"] 
+    if len(user_data["watched"]) == 0:
+        return 0.0 
+
+    average_rating = sum_counter/len(user_data["watched"])
+    return average_rating 
+
+    # PROBLEM TWO 
+    # iterating through the watched list
+    # each loop gives us one movie 
+    # get genre from movie 
+    # add new genre into the dictionary as a key with count being the value, (check if genre is in the dict and add one) OR (add too dict at 1)
+    # Now iterate over different_genre dict
+    # Find highest value  
+    # return key to highest value 
+def get_most_watched_genre(user_data):
+    if not user_data["watched"]:
+        return None
+
+    genres = {}
+    for movie in user_data["watched"]:
+        for i, x in movie.items():
+            if i == "genre":
+                if x in genres:
+                    genres[x] += 1 
+                else:
+                    genres[x] = 1
+
+    count = 0
+    most_watched = ""
+
+    for i, x in genres.items():
+        if x > count:
+            count = x
+            most_watched = i 
+        
+    return most_watched
+
+
+
+
+# WAVE THREE 
+
+    # PROBLEM ONE 
+def get_unique_watched(user_data):
+    unique_movies = []
+    have_seen = {}
+    friends = user_data["friends"] # list of friends 
+    for movie in user_data["watched"]:
+        have_seen[movie["title"]] = False 
+    
+    for friend in friends:
+        friend_seen_list = friend["watched"]
+        for friend_movie in friend_seen_list:
+            friend_movie_title = friend_movie["title"]
+            if friend_movie_title in have_seen:
+                have_seen[friend_movie_title] = True
```

Comment:
This is a clever use of a dictionary to store `True` and `False` values to be used later. Notice that you almost have this exact same code below in `get_friends_unique_watched`. Consider moving this code to a helper method to DRY up your code.